core/src/main/scala/org/genivi/sota/core/CampaignLauncher.scala

```diff
@@ -0,0 +1,48 @@
+/**
+  * Copyright: Copyright (C) 2016, ATS Advanced Telematic Systems GmbH
+  * License: MPL-2.0
+  */
+package org.genivi.sota.core
+
+import akka.actor.ActorSystem
+import cats.Foldable
+import cats.implicits._
+import org.genivi.sota.core.data.Campaign
+import org.genivi.sota.core.db.Campaigns
+import org.genivi.sota.common.DeviceRegistry
+import org.genivi.sota.data.{Namespace, PackageId, Uuid}
+import scala.concurrent.{ExecutionContext, Future}
+import slick.driver.MySQLDriver.api.Database
+
+object CampaignLauncher {
+  import UpdateService._
+
+  private def resolve(devices: Seq[Uuid]): DependencyResolver = { pkg =>
+    val elems = devices.map(_ -> Set(pkg.id))
+    Future.successful(Map(elems:_*))
+  }
```

Comment:
So this is what @jerrytrieu told me, we will probably use the resolver in the future, but for now the campaigns should work only with the groups.